### PR TITLE
Use the `ApplicationIcon` as the `Form.DefaultIcon`

### PR DESCRIPTION
**Pull request by elachlan**
Test implementation based on discussion.

Related: #8905 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8918)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
@elachlan We probably should add new API to `Icon` to wrap getting icons from the native resource in a PE file. I'll create an API proposal for it. It fits with the theme of changes for stock icons.

**Comment by elachlan** [comment]
That would be great. It would make this a lot easier. The PInvoke function is slightly difficult. Happy to help with the API change. Could you CC me in that issue once you create it?

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 1]
```diff
@@ -897,7 +897,9 @@ internal static Icon DefaultIcon
                     {
                         // Once we grab the lock, we re-check the value to avoid a
                         // race condition.
-                        defaultIcon ??= new Icon(typeof(Form), "wfc");
+                        defaultIcon ??=
+                            PInvoke.ExtractIconEx(Process.GetCurrentProcess().MainModule?.FileName)
```
Comment (elachlan) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs:901]: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1839

Needs to use `Environment.ProcessPath` instead

**Comment by elachlan** [comment]
@JeremyKuhne I will refactor this once your `Icon` changes are merged.

**Comment by JeremyKuhne** [comment]
@elachlan https://github.com/dotnet/winforms/issues/8905#issuecomment-1506130790

**Comment by dreddy-work** [comment]
@elachlan, seems this is waiting on you.

**Comment by elachlan** [comment]
I will close this as we can't currently implement it correctly because of limitations in icon.

**Pull request closed without merging**